app/javascript/components/history/History.js

```diff
@@ -1,31 +1,134 @@
-  );
-};
+class History extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      edit_mode: false,
+      loading: false,
+      comment: '',
+    };
+  }
+
+  handleEditComment = () => {
+    this.setState({ edit_mode: true });
+  };
+
+  handleTextChange = event => {
+    this.setState({ comment: event.target.value });
+  };
+
+  handleDeleteComment = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/histories/' + this.props.history.id)
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  formatEditable = () => {
+    if (this.props.history.history_type == 'Comment') {
+      return (
+        <React.Fragment>
+          <Button variant="link" id={this.props.history.id} className="py-0" onClick={this.handleEditComment}>
+            Edit
+          </Button>
+          <Button variant="link" id={this.props.history.id} className="py-0" onClick={this.handleDeleteComment}>
+            Delete
+          </Button>
+        </React.Fragment>
+      );
+    } else {
+      return '';
+    }
+  };
+
+  editComment = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .patch(window.BASE_PATH + '/histories/' + this.props.history.id, {
+          comment: this.state.comment,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  renderEditMode = () => {
+    if (this.state.edit_mode == true) {
+      return (
+        <Card.Body>
+          <textarea
+            id="comment"
+            name="comment"
+            className="form-control"
+            style={{ resize: 'none' }}
+            rows="3"
+            placeholder={this.props.history.comment}
+            value={this.state.comment}
+            onChange={this.handleTextChange}
+          />
+          <button className="mt-3 btn btn-primary btn-square float-right" disabled={this.state.loading || this.state.comment === ''} onClick={this.editComment}>
+            <i className="fas fa-comment-dots"></i> Edit Comment
+          </button>
+        </Card.Body>
+      );
+    } else {
+      return (
+        <Card.Body>
+          <Card.Text>{this.props.history.comment}</Card.Text>
+        </Card.Body>
+      );
+    }
+  };
 
-history.propTypes = {
+  render() {
+    return (
+      <React.Fragment>
+        <Card className="card-square mt-4 mx-3 shadow-sm">
+          <Card.Header>
+            <b>{this.props.history.created_by}</b>, {time_ago_in_words(moment(this.props.history.created_at).toDate())} ago (
+            {moment
+              .tz(this.props.history.created_at, 'UTC')
```

Comment:
An additional thought, can we move this time_ago_in_words method (plus the ago text) into the /javascript/utils/DateTime.js and call it `formatRelativeDate`? @DPC15038 started breaking out our util methods into one place and I think this would be a good candidate